src/astro/sql/table.py

```diff
@@ -49,22 +53,26 @@ def __post_init__(self):
         if not self._name:
             self.temp = True
 
-    def _create_unique_table_name(self) -> str:
+    def _create_unique_table_name(self, prefix: str = "") -> str:
         """
         If a table is instantiated without a name, create a unique table for it.
         This new name should be compatible with all supported databases.
         """
         schema_length = len((self.metadata and self.metadata.schema) or "") + 1
+        prefix_length = len(prefix)
+
         unique_id = random.choice(string.ascii_lowercase) + "".join(
             random.choice(string.ascii_lowercase + string.digits)
-            for _ in range(MAX_TABLE_NAME_LENGTH - schema_length)
+            for _ in range(MAX_TABLE_NAME_LENGTH - schema_length - prefix_length)
         )
+        if prefix:
+            unique_id = f"{prefix}{unique_id}"
+
         return unique_id
 
-    # Due to some weird issue in python typing, I can not actually record that this function returns a table
```

Comment:
Fixed it in https://github.com/astronomer/astro-sdk/pull/384/commits/13b68de9ccbe00ffebd3ed64b256c03bce48741a , it is because the return type is actually the part of the Class definition itself.

>When a type hint contains names that have not been defined yet, that definition may be expressed as a string literal, to be resolved later.

Reference: https://peps.python.org/pep-0484/#forward-references

Reply:
@kaxi I understand the problem, and I solved it by importing `from __future__ import annotations`, which will be the default behavior starting in Python 3.11. The downside with this approach was that Black automatically converted several types of annotations.

Reply:
@kaxil I think I know what happened. When I added:
```
 from __future__ import annotations
```
One of the pre-commit checks said that the imports from typing were no longer necessary. 

When I removed those imports, black changed the syntax for type checking.